### DISCUSSION TITLE
Prisma Security in React Native Expo App

### DISCUSSION THREAD

**Discussion by todayssky** [supabase/supabase#27973] [Questions] [answered]
I am trying to use Prisma with a Supabase database in a React Native Expo app without any backend code. However, to use Prisma, I need to include the database URL (containing the database ID and password) in the environment variables. I understand that this URL could be exposed in the mobile app. Is this acceptable? If not, what are the alternatives?

Also, is it not possible to use transaction functionality when using supabase-js in React Native? If so, how can I use the transaction functionality?

Thanks!!!

**Comment by rossnoah** [comment]
This clearly unacceptable from a security standpoint, the alternative is to stand up an API, or use the built in Supabase postgREST API. Either way make sure you enable RLS and configure it as otherwise anyone can access all the data in your database.

**Comment by GaryAustin1** [comment, marked as answer]
Prisma only uses the direct database port and as you guess and other user replies you can not expose that.

The only way to do transactions with the REST API is to use an rpc call to a Postgres function.
